seqr/utils/search/hail_search_utils.py

```diff
@@ -87,6 +87,9 @@ def _execute_lookup(variant_id, data_type,  user, **kwargs):
 
 
 def hail_variant_lookup(user, variant_id, **kwargs):
+    if kwargs.get('genome_version') == 'GRCh37' and variant_id[0] == 'M':
+        from seqr.utils.search.utils import InvalidSearchException
+        raise InvalidSearchException('Mitochondrial variants are not supported in GRCh37')
     variant, _ = _execute_lookup(variant_id, Sample.DATASET_TYPE_VARIANT_CALLS, user, **kwargs)
```

Comment:
So it looks like you are checking here if the variant starts with"M", but we are still sending the lookup query with the wrong data type. I think the solution here actually needs to involve correctly setting the data type for this search, and then validating it. The `_variant_ids_dataset_type` helper function in the main search utils.py is probably a good place to share logic. You can also keep the error handling in this file or moe it into the other file, whichever is easier

Reply:
Also, we should probably add this validation to SV lookup as well

Reply:
right, the default of Sample.DATASET_TYPE_VARIANT_CALLS isn't correct for MITO. moving the validation elsewhere makes sense

Reply:
@hanars made some changes, lmk if this is more in line with what you're thinking